cloud/types_preview.go

```diff
@@ -42,8 +45,36 @@ type (
 		PreviewID string                `json:"preview_id"`
 		Stacks    ResponsePreviewStacks `json:"stacks"`
 	}
+
+	// UpdateStackPreviewPayloadRequest is the request payload for the update of
+	// stack previews.
+	UpdateStackPreviewPayloadRequest struct {
+		Status           string            `json:"status"`
+		ChangesetDetails *ChangesetDetails `json:"changeset_details,omitempty"`
+	}
 )
 
+// Validate the UpdateStackPreviewPayloadRequest object.
+func (r UpdateStackPreviewPayloadRequest) Validate() error {
+	errs := errors.L()
+	if r.Status == "" {
+		errs.Append(errors.E("status is required"))
+	}
+
+	if r.ChangesetDetails != nil {
+		if r.ChangesetDetails.Provisioner == "" {
+			errs.Append(errors.E("provisioner is required"))
+		}
+		if r.ChangesetDetails.ChangesetASCII == "" {
+			errs.Append(errors.E("changeset_ascii is required"))
+		}
+		if r.ChangesetDetails.ChangesetJSON == "" {
+			errs.Append(errors.E("changeset_json is required"))
+		}
+	}
```

Comment:
Only `provisioner` and one of "changeset_ascii" or "changeset_json" is required.
If terramate successfully generates the ASCII version but fail to generate the JSON, it logs a warn and proceeds with just the ASCII.

Additionally, the `ChangesetDetails` type already has validation implemented, can't you use it here?